File: apps/backend/app/models/time_entry.py
import uuid
from sqlalchemy import Column, String, DateTime, ForeignKey, Integer
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.session import Base
from sqlalchemy.dialects.postgresql import UUID


class TimeEntry(Base):
    __tablename__ = "time_entries"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    task_id = Column(UUID(as_uuid=True), ForeignKey("tasks.id"), nullable=False)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    start_time = Column(DateTime(timezone=True), nullable=False)
    end_time = Column(DateTime(timezone=True), nullable=True)
    duration_minutes = Column(Integer, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    task = relationship("Task", back_populates="time_entries")
    user = relationship("User")

File: apps/frontend/src/types/time_entry.ts
export interface TimeEntry {
    id: string;
    card_id: string;
    user_id?: string | null;
    description?: string | null;
    start_time?: string | null;
    end_time?: string | null;
    duration_minutes: number;
    created_at: string;
}

export interface CreateTimeEntryData {
    description?: string;
    duration_minutes?: number;
    start_time?: string;
    end_time?: string;
}

export interface TimeEntriesListResponse {
    entries: TimeEntry[];
    total_minutes: number;
}
